src/univariate/continuous/triangular.jl

```diff
@@ -119,27 +133,22 @@ function quantile(d::TriangularDist, p::Real)
               b - sqrt(b_m_a * (b - c) * (1 - p))
 end
 
-function mgf(d::TriangularDist{T}, t::Real) where T<:Real
-    if t == zero(t)
-        return one(T)
-    else
-        (a, b, c) = params(d)
-        u = (b - c) * exp(a * t) - (b - a) * exp(c * t) + (c - a) * exp(b * t)
-        v = (b - a) * (c - a) * (b - c) * t^2
-        return 2u / v
-    end
+function mgf(d::TriangularDist, t::Real)
+    a, b, c = params(d)
+    T = typeof(t / a)
+    t == zero(T) && return one(T)
```

Comment:
This can be written simpler as
```suggestion
    iszero(t) && return one(T)
```

Reply:
ok, I took your suggestion